ggircs-app/app/components/FuelManagement/CarbonTaxActFuelType.tsx

```diff
@@ -0,0 +1,202 @@
+            <FuelSelectionComponent
+              queryParameter="carbonTaxActFuelTypeId"
+              displayParameter="carbonTaxFuelType"
+              data={allCarbonTaxActFuelTypes?.allCarbonTaxActFuelTypes}
+              pageQuery={pageQuery}
+            />
+          </Card>
+        </Col>
+        <Col md="9">
+          {carbonTaxActFuelType && (
+            <>
+              <Row>
+                <Col md="12">
+                  <Card>
+                    <Card.Header className="bc-card-header">
+                      <Card.Title>
+                        {carbonTaxActFuelType.carbonTaxFuelType} (
+                        {carbonTaxActFuelType.ctaRateUnits})
+                      </Card.Title>
+                    </Card.Header>
+                    <Card.Body>
+                      <Button
+                        id="show-hide-toggle"
+                        variant="secondary"
+                        onClick={handleShowCharges}
+                      >
+                        {showPeriodText}
+                      </Button>
+                      <Table responsive striped bordered hover size="small">
+                        <thead>
+                          <tr>
+                            <th>Period Start</th>
+                            <th>Period End</th>
+                            <th>Charge</th>
+                            <th>Comments</th>
+                            <th>Edit</th>
+                          </tr>
+                        </thead>
+                        <tbody>
+                          {carbonTaxActFuelType.fuelChargesByCarbonTaxActFuelTypeId.edges
+                            .filter((node, idx) => idx >= fromIndex)
+                            .map(({ node }) => (
+                              <FuelChargeRow
+                                key={node.id}
+                                fuelId={carbonTaxActFuelType.rowId}
+                                fuelCharge={node}
+                                validateRatePeriod={validateRatePeriod}
+                              />
+                            ))}
+                          <CreateEditFuelChargeRow
+                            operation={"create"}
+                            fuelId={carbonTaxActFuelType.rowId}
+                            connectionId={
+                              carbonTaxActFuelType
+                                .fuelChargesByCarbonTaxActFuelTypeId.__id
+                            }
+                            validateRatePeriod={validateRatePeriod}
+                          />
+                        </tbody>
+                      </Table>
+                    </Card.Body>
+                  </Card>
+                </Col>
+              </Row>
+            </>
+          )}
+          {!carbonTaxActFuelType && (
+            <Row>
+              <Col>
+                Please select a Carbon Tax Act Fuel on the left to view details
+                and related fuel charges
```

Comment:
Are we calling them fuel charges or rates? I think that the CTA uses "rate", so we might want to align with it there